### PR TITLE
fix: apply default color and background for sheet elements

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Add SASS mixin for default background/color of components

## Motivation and Context
Normalize component background/color defaults

## How Has This Been Tested?

visually

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-container>
      <div class="text-h5 ma-2">v-btn</div>
      <template v-for="variant in ['contained', 'outlined', 'text', 'plain']" :key="variant">
        <div class="mb-12">
          <v-btn class="mr-6" :variant="variant" link to="/page1">
            {{ variant }} variant
          </v-btn>

          <v-btn color="primary" :variant="variant" link>
            {{ variant }} variant
          </v-btn>
        </div>
      </template>

      <div class="text-h5 ma-2">v-list</div>
      <template v-for="variant in ['contained', 'outlined', 'text', 'plain']" :key="variant">
        <v-list :subheader="variant + ' variant'">
          <v-list-item title="Default" to="/page2" :variant="variant" />
          <v-list-item title="Colored" link :variant="variant" color="primary" />
        </v-list>
      </template>

      <div class="text-h5 ma-2 my-12">v-card</div>

      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dolorem enim id corporis libero eveniet quibusdam sit quam eligendi magnam. Dicta aut recusandae sint natus soluta corrupti necessitatibus perspiciatis nulla eos.

      <template v-for="variant in ['contained', 'outlined', 'text', 'plain']" :key="variant">
        <div class="mb-12">
          <v-card class="mb-6" :variant="variant" link>
            <v-card-header>
              <v-card-header-text>
                <v-card-title>{{ variant }} variant</v-card-title>

                <v-card-subtitle>Default</v-card-subtitle>
              </v-card-header-text>
            </v-card-header>

            <v-card-text>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quaerat aperiam hic quo aspernatur nihil quidem corporis asperiores numquam repudiandae tempore, ad voluptates non deleniti.
            </v-card-text>
          </v-card>

          <v-card color="primary" :variant="variant" link>
            <v-card-header>
              <v-card-header-text>
                <v-card-title>{{ variant }} variant</v-card-title>

                <v-card-subtitle>Colored</v-card-subtitle>
              </v-card-header-text>
            </v-card-header>

            <v-card-text>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quaerat aperiam hic quo aspernatur nihil quidem corporis asperiores numquam repudiandae tempore, ad voluptates non deleniti.
            </v-card-text>
          </v-card>
        </div>
      </template>
    </v-container>
  </v-app>
</template>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VAppBar/_variables.scss [hunk 1]
```diff
@@ -30,3 +31,9 @@ $app-bar-border: (
   $app-bar-border-style,
   $app-bar-border-width
 ) !default;
+
+$app-bar-theme: () !default;
+$app-bar-theme: (
```
Comment (KaelWD) [inline packages/vuetify/src/components/VAppBar/_variables.scss:36]: Why declare it twice?

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/core-team

**Comment by KaelWD** [comment]
This breaks non-contained buttons (they shouldn't have a background)
image

**Comment by KaelWD** [comment]
And you missed VCard

File: packages/vuetify/src/components/VBtn/_variables.scss [hunk 2]
```diff
@@ -27,6 +27,7 @@ $button-border-width: 0 !default;
 $button-card-actions-elevation: 0 !default;
 $button-card-actions-margin: .5rem !default;
 $button-card-actions-padding: 0 8px !default;
+$button-color: rgb(var(--v-theme-on-surface)) !default;
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBtn/_variables.scss:30]: Duplicated

File: packages/vuetify/src/components/VBadge/VBadge.sass [hunk 3]
```diff
@@ -5,6 +5,8 @@
   display: inline-block
   line-height: $badge-line-height
 
+  @include theme($badge-theme...)
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBadge/VBadge.sass:8]: Shouldn't it just apply to the badge "dot"

Comment (johnleider) [reply packages/vuetify/src/components/VBadge/VBadge.sass:8]: I agree. We should probably just integrate variants into badge later on.

**Pull request merged**